pennylane/ops/functions/iterative_qpe.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.iterative_qpe function.
+"""
+
+import numpy as np
+import pennylane as qml
+
+
+def iterative_qpe(base, estimation_wire, iters):
+    r"""Performs the `iterative quantum phase estimation <https://arxiv.org/pdf/quant-ph/0610214.pdf>`_ circuit.
+
+    Given a unitary matrix :math:`U`, this function applies the circuit for iterative quantum phase
+    estimation and returns a list of the mid-circuit measurements.
+
+    Args:
+      base (Operator): the phase estimation unitary, specified as an :class:`~.Operator`
+      estimation_wire (Union[Wire, int, str]): the wire to be used for the estimation.
+      iters (int): the number of measurements to be performed
+
+    Returns:
+      list[MidMeasureMP]: the list of measurements performed
+
+    .. seealso:: :class:`~.QuantumPhaseEstimation`, :func:`~.measure`
+
+    **Example**
```

Comment:
Maybe add a sentence explaining how to interpret the output. Also I think the PR which fixes mid-circuit measurement drawing has been merged? So we might be able to draw the circuit for this example as well.

Reply:
That PR does not consider `qml.cond` yet

Reply:
Yes, I'm in the process of adding support for `qml.cond`. Currently there's an open PR adding support for multiple `qml.cond` relying on the same (single) mid-circuit measurement, but that wouldn't be enough for drawing iterative QPE

Reply:
added example :)